### PR TITLE
feat(organization-invites): handle duplicate invites

### PR DESCRIPTION
**Pull request by raquelmsmith**
## Problem

If someone is using the API to create invites, it's possible to create multiple invites for the same user. This is _fine_ but not ideal, esp if someone is using the `private_project_access` field on the invites to specify projects someone should get access to when they join.

## Changes

When creating an invite, the `combine_pending_invites` field can be optionally applied. 
- If True, non-expired pending invoices are combined, getting the highest access level specified on any invite and using those for the resulting invite. Old invites are deleted.
- If False (default), just delete the old invite and create a new one with the current invite info being supplied via the request.

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

## Does this work well for both Cloud and self-hosted?

Should!

## How did you test this code?

Added a few tests.

### CODE REVIEW

File: posthog/api/organization_invite.py [hunk 1]
```diff
@@ -107,12 +111,65 @@ def create(self, validated_data: dict[str, Any], *args: Any, **kwargs: Any) -> O
...
+
+            validated_data["private_project_access"] = [
+                {"id": project_id, "level": level} for project_id, level in combined_access.items()
+            ]
+
+        # Delete existing invites
+        existing_invites.delete()
+
+        # delete any existing expired invites
+        expired_invites = OrganizationInvite.objects.filter(
```
[the remark this answers is not in the document]
Comment (raquelmsmith) [reply posthog/api/organization_invite.py:160]: Just for cleanliness... why _wouldn't_ we do this? If only the most recent invite gets consumed, why keep the older expired ones around?

[the remark this answers is not in the document]
Comment (zlwaterfield) [reply posthog/api/organization_invite.py]: It just feels weird to expire all invites on the creation of a new one but the logic itself makes sense.

[the remark this answers is not in the document]
Comment (zlwaterfield) [reply posthog/api/organization_invite.py]: I also think expired invites are nice to have, I don't like deleting them unless another one is made so it's clear that one was sent but not accepted.

**Review by zlwaterfield — APPROVED** [review]
Looks good, I don't love the deleting of all invites but not a blocker for me.

File: posthog/api/organization_invite.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (raquelmsmith) [reply posthog/api/organization_invite.py]: Oh, this should only delete them when a new one is made, unless something in my logic is incorrect?

[the remark this answers is not in the document]
Comment (zlwaterfield) [reply posthog/api/organization_invite.py]: The original commit had this so it deleted all expired invites for the org no matter what the email was. 
```
		expired_invites = OrganizationInvite.objects.filter(
			organization_id=self.context["organization_id"],
            created_at__lt=datetime.now() - timedelta(days=INVITE_DAYS_VALIDITY),
        )
        expired_invites.delete()
```
It looks like the latest no longer has this.

[the remark this answers is not in the document]
Comment (raquelmsmith) [reply posthog/api/organization_invite.py]: Oh! That would have been unintended, so nice catch! Thanks!

[the remark this answers is not in the document]
Comment (zlwaterfield) [reply posthog/api/organization_invite.py]: Ah ok cool. Well we are all good then :)

**Pull request merged**